iset.mm

```diff
@@ -64252,6 +64252,15 @@ all reals (greatest real number) for which all positive reals are greater.
       UDUMUGUJBMSHBQAUETUF $.
   $}
 
+  ${
+    $( Left and right injection never produce equal values.  (Contributed by
+       Jim Kingdon, 2-Jul-2022.) $)
+    djune $p |- ( ( A e. V /\ B e. W ) -> ( inl ` A ) =/= ( inr ` B ) ) $=
```

Comment:
Not sure how you want to use it, but similarly, I would propose
```
$p |- ( ( A e. V /\ B e. W ) -> ( ( inl |` V ) ` A ) =/= ( ( inr |` W ) ` B ) ) $=
```

Reply:
I suppose that would work with some extra steps (with V and W set to `_V` for the case I care about), but I'm not sure I see the point, especially when V and W are just arbitrary classes, not tethered to any disjoint union.

To explain how it is being used, it is part of the formalization of:

> Suppose g(0) = inr(0) and g(1) = inr(0). We prove ¬p holds. To this end, assume p holds.
Hence 0 ∈ A. Since g is surjective there must be some i ∈ 2 such that g(i) = inl(0),
contradicting our assumption.

from the PradicBrown2021 proof.

The proof in #2659 formalizes this by showing that ``( inl ` (/) ) = ( inr ` (/) )`` (because they are both equal to g(i) as described in the quote), and then ``( inl ` (/) ) =/= ( inr ` (/) )`` by djune which is the contradiction described in the quote.

Reply:
As above: they actually use the functions `` ( inl |` A ) `` and `` ( inr |` 1o ) ``. Every left-injection is from some X to some X |_| Y, so is of the form ``( inl |` X )``. Actually, `inl` is a "function template".